src/Widgets/Canvas.vala

```diff
@@ -123,17 +124,22 @@ public class GtkCanvas.Canvas : Gtk.AspectFrame {
         add (stage);
     }
 
-   /**
+    /**
     * Adds a test shape. Great for testing the library!
     *
+    * @param type the shape to be generated [rectangle, circle, triangle]
     * @param color the color the test-shape will be, in CSS format
     * @param rotation the amount of degrees the item will be rotated
     */
-    public CanvasItem add_test_shape (string color, double rotation) {
-        var item = new CanvasItem ();
-        item.background_color = Clutter.Color.from_string (color);
-
-        item.rotation = rotation;
+    public CanvasItem add_shape (string type, string color, double rotation) {
+        switch (type) {
```

Comment:
item should be declared here, on top of switch :)

Reply:
If I do that, I get an error while compiling.
```
Canvas.vala:145.9-145.24: error: use of possibly unassigned local variable `item'
        add_item (item);
```

Reply:
You can always null-check 😉

Reply:
Or instead of a null check, you can have a `default:` on the switch :)

Reply:
:+1: